core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -1849,6 +1849,17 @@ public static <T extends Comparable<T>> T greatest(T b0, T b1) {
     return b0 == null || b1 != null && b0.compareTo(b1) < 0 ? b1 : b0;
   }
 
+  /** Less than. */
+  public static <T extends Comparable<T>> boolean lessThan(T b0, T b1) {
+    return b1 == null || b0 != null && b0.compareTo(b1) < 0;
+  }
+
+  /** Grater than. */
+  public static <T extends Comparable<T>> boolean greaterThan(T b0, T b1) {
+    return b1 == null || b0 != null && b0.compareTo(b1) > 0;
```

Comment:
What's the difference between `greatest` and `greaterThan`?

Reply:
`greaterThan` return wether b0 is greater than b1 while `greatest` return the larger one between b0 and b1